app/controllers/resources_controller.rb

```diff
@@ -1,14 +1,22 @@
 class ResourcesController < ApplicationController
   def index
     category_id = params[:category_id]
+    latitude = params[:latitude]
+    longitude = params[:longitude]
+
     result = if category_id
                resources.joins(:categories).where('categories.id' => category_id)
              end
 
     if result
+      if latitude and longitude
+        result = result.sort_by{ |resource| resource.address.distance_to([latitude,longitude])}
```

Comment:
ok yea that makes sense, for some reason I thought I wasn't able to include it in the db query, ill try this out